pkg/kubernetes/volume/v1alpha1/build.go

```diff
@@ -64,6 +64,32 @@ func (b *Builder) WithHostDirectory(path string) *Builder {
 	b.volume.object.VolumeSource = volumeSource
 	return b
 }
+func (b *Builder) WithConfigMap(configMap *corev1.ConfigMap, defaultmode int32) *Builder {
+	if configMap == nil {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build volume object: nil ConfigMap"),
+		)
+		return b
+	}
+	if defaultmode == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build volume object: missing defaultmode"),
+		)
+	}
+	volumeSource := corev1.VolumeSource{
+		ConfigMap: &corev1.ConfigMapVolumeSource{
+			DefaultMode: &defaultmode,
```

Comment:
Assigning an address in this way is error-prone i.e If caller code modifies `defaultmode` after making a call to this function then it will cause a problem. Please make a local copy and use it.